dali/operators/decoder/nvjpeg/nvjpeg_decoder_decoupled_api.h

```diff
@@ -39,6 +39,10 @@
 #include "dali/core/static_switch.h"
 #include "dali/operators/decoder/nvjpeg/permute_layout.h"
 
+#define IF_HW_DECODER_COMPATIBLE \
+        NVJPEG_VER_MAJOR > 11 || \
+        (NVJPEG_VER_MAJOR == 11 && (NVJPEG_VER_MINOR > 4 || \
+                                   (NVJPEG_VER_MINOR == 4 && NVJPEG_VER_PATCH >= 1)))
```

Comment:
Normally when one defines such macro, the condition would be in parenthesis - otherwise there can be "interesting" results, like:
````#if IF_HW_DECODER_COMPATIBLE && something``` would resolve to:
```
#if NVJPEG_VER_MAJOR > 11 || \
        (NVJPEG_VER_MAJOR == 11 && (NVJPEG_VER_MINOR > 4 || \
                                   (NVJPEG_VER_MINOR == 4 && NVJPEG_VER_PATCH >= 1))) && something
```
The `&& something` only applies to the right hand side of `||`.
Therefore, it should be:
```suggestion
#define (IF_HW_DECODER_COMPATIBLE \
        NVJPEG_VER_MAJOR > 11 || \
        (NVJPEG_VER_MAJOR == 11 && (NVJPEG_VER_MINOR > 4 || \
                                   (NVJPEG_VER_MINOR == 4 && NVJPEG_VER_PATCH >= 1))))
```
or
```suggestion
#if NVJPEG_VER_MAJOR > 11 || \
        (NVJPEG_VER_MAJOR == 11 && (NVJPEG_VER_MINOR > 4 || \
                                   (NVJPEG_VER_MINOR == 4 && NVJPEG_VER_PATCH >= 1)))
#define IF_HW_DECODER_COMPATIBLE 1
#else
#define IF_HW_DECODER_COMPATIBLE 0
#endif
```

Reply:
Done